EIPS/eip-5089.md

```diff
@@ -0,0 +1,393 @@
+---
+eip: 5089
+title: Principal Token
+description: Principal tokens (zero-coupon tokens) are redeemable for a single underlying EIP-20 token at a future timestamp.
+author: Julian Traversa (@JTraversa), Robert Robbins (@robrobbins), Alberto Cuesta Cañada (@alcueca)
+discussions-to: https://ethereum-magicians.org/t/eip-5089-principal-token-standard/9259
+status: Draft
+type: Standards Track
+category: EIP
+created: 2022-05-01
+requires: 20, 2612
+---
+
+## Abstract
+
+Principal tokens represent ownership of an underlying [EIP-20](./eip-20.md) token at a future timestamp.
+
+This specification is an extension on the [EIP-20](./eip-20.md) token that provides basic functionality for depositing
+and withdrawing tokens and reading balances and the [EIP-2612](./eip-2612.md) specification that provides
+[EIP-712](./eip-712.md) signature based approvals.
+
+## Motivation
+
+Principal tokens lack standardization which has led to a difficult to navigate development space and diverse implementation
+schemes.
+
+The primary examples include yield tokenization platforms which strip future yield leaving a principal
+token behind, as well as fixed-rate money-markets which utilize principal tokens as a medium
+to lend/borrow.
+
+This inconsistency in implementation makes integration difficult at the application layer as well as
+wallet layer which are key catalysts for the space's growth. 
+Developers are currently expected to implement individual adapters for each principal token, as well as adapters for
+their pool contracts, and many times adapters for their custodial contracts as well, wasting significant developer resources. 
+
+## Specification
+
+All Principal Tokens (PTs) MUST implement [EIP-20](./eip-20.md) to represent ownership of future underlying redemption.
+If a PT is to be non-transferrable, it MAY revert on calls to `transfer` or `transferFrom`.
+The [EIP-20](./eip-20.md) operations `balanceOf`, `transfer`, `totalSupply`, etc. operate on the Principal Token balance.
+
+All Principal Tokens MUST implement [EIP-20](./eip-20.md)'s optional metadata extensions.
+The `name` and `symbol` functions SHOULD reflect the underlying token's `name` and `symbol` in some way, as well as the origination protocol, and in the case of yield tokenization protocols, the origination money-market.
+
+All Principal Tokens MAY implement [EIP-2612](./eip-2612.md) to improve the UX of approving PTs on various integrations.
+
+### Definitions:
+
+- underlying: The token that Principal Tokens are redeemable for at maturity.
+  Has units defined by the corresponding [EIP-20](./eip-20.md) contract.
+- maturity: The timestamp (unix) at which a Principal Token matures. Principal Tokens become redeemable for underlying at or after this timestamp.
+- fee: An amount of underlying or Principal Token charged to the user by the Principal Token. Fees can exist on redemption or post-maturity yield.
```

Comment:
Are there any examples of contracts like these that charge fees on redemption? Might it be reasonable to enforce that fees are only charged during issuance/mint?

Reply:
I can't think of any examples, but I don't like to enforce business models in a standard unless there is a good technical reason for it. What would we get out of banning fees on redemption?

Reply:
Yea, this would only justified in the context of my other comment around enforcing a `1:1` intended redemption at maturity between PTs and the underlying